src/rules/rule.ts

```diff
@@ -0,0 +1,42 @@
+const BLOCK_RULE = 'block';
+const APPROVE_RULE = 'approve';
+
+/** 
+ * Rule class
+ * Each rule have a name and a callback method that check 
+ * if the rule match/dismatch
+ **/
+export class Rule {
+    func: Function;
```

Comment:
You want to be sure here you receive not just any function but function which receive one parameter and need to return `boolean`. I wood replace `Function` by (req: any) => boolean